Snowflake.Data/Core/SFBlockingChunkDownloaderV3.cs

```diff
@@ -184,25 +181,15 @@ private async Task<IResultChunk> DownloadChunkAsync(DownloadContextV3 downloadCo
                         {
                             await ParseStreamIntoChunk(stream, chunk);
                         }
-                        if (forceParseError)
-                        {
-                            throw new Exception("json parsing error.");
-                        }
                     }
                     catch (Exception e)
                     {
-                        forceParseError = false;
                         if (retryCount < HttpUtil.MAX_RETRY)
                         {
                             retry = true;
                             // reset the chunk before retry in case there could be garbage
                             // data left from last attempt
-                            ExecResponseChunk chunkInfo = new ExecResponseChunk()
-                            {
-                                rowCount = chunk.RowCount,
-                                url = chunk.Url
-                            };
-                            chunk.Reset(chunkInfo, chunk.chunkIndexToDownload);
+                            chunk.Reset(chunkInfos[chunk.chunkIndexToDownload], chunk.chunkIndexToDownload);

```

Comment:
Thanks @sfc-gh-mhofman for catching the issue here. I agree uncompressedSize needs to be reset correctly as well (although I'm not sure if it would cause customer facing issue as the test passed previously).
I added the reset here to fix the issue that when chunk downloading failed in the middle, without reset the chunk the data left from the failed attempt will break all following retry attempts.
My concern of using chunkInfos here is it was used in main thread only before and the existing implementation carefully not passing chunk, not chunk index to the downloading thread.
https://github.com/snowflakedb/snowflake-connector-net/blob/v2.0.24/Snowflake.Data/Core/SFBlockingChunkDownloaderV3.cs#L113
It's possible that application stop fetching in the middle and close the connection, while downloading thread still working. In such case if downloading thread try to access the resource in ChunkDownloader, which could be destroyed already, in C++ it would crash the driver for sure, but I'm not sure what would happen in .NET. If that's not a problem in .NET the current fix should be fine, otherwise maybe adding a new function in ExecResponeChunk like Reset4Retry(), simply reset internal data buffer using existing uncompressedSize.

Reply:
LGTM